### ISSUE TITLE
fortios_firewall_shapingpolicy requiring service and dstaddr blocks

### ISSUE THREAD

**Issue by zippanto** [fortinetdev/terraform-provider-fortios#306]
Hi,

When trying to define a fortios_firewall_shapingpolicy resource with internet_service = "enable" and specifying internet_service_name we are getting an error during plan/apply:

```
│ Error: Insufficient service blocks
│ 
│   on traffic_shaping.tf line 14, in resource "fortios_firewall_shapingpolicy" "firewall_shapingpolicy":
│   14: resource "fortios_firewall_shapingpolicy" "firewall_shapingpolicy" {
│ 
│ At least 1 "service" blocks are required.
╵
╷
│ Error: Insufficient dstaddr blocks
│ 
│   on traffic_shaping.tf line 14, in resource "fortios_firewall_shapingpolicy" "firewall_shapingpolicy":
│   14: resource "fortios_firewall_shapingpolicy" "firewall_shapingpolicy" {
│ 
│ At least 1 "dstaddr" blocks are required.
```

We tried importing the resource, running terraform state show to then re-use the output (please see below for the output) of the command in the plan. However we are still getting the same error message on plan/apply.

```
% terraform state show 'fortios_firewall_shapingpolicy.firewall_shapingpolicy["phones"]'
# fortios_firewall_shapingpolicy.firewall_shapingpolicy["phones"]:
resource "fortios_firewall_shapingpolicy" "firewall_shapingpolicy" {
    class_id               = 0
    comment                = "phones"
    diffserv_forward       = "disable"
    diffserv_reverse       = "disable"
    diffservcode_forward   = "000000"
    diffservcode_rev       = "000000"
    dynamic_sort_subtable  = "false"
    fosid                  = 19
    get_all_tables         = "false"
    id                     = "19"
    internet_service       = "enable"
    internet_service_src   = "disable"
    ip_version             = "4"
    name                   = "phones"
    status                 = "enable"
    tos                    = "0x00"
    tos_mask               = "0x00"
    tos_negate             = "disable"
    traffic_shaper         = "phones-high-priority"
    traffic_shaper_reverse = "phones-high-priority"

    dstintf {
        name = "virtual-wan-link"
    }

    internet_service_name {
        name = "8X8-8X8.Cloud"
    }

    srcaddr {
        name = "phones"
    }
}
```

If we add dstaddr and service blocks with name = "dummy" the error goes away, however this workaround is obviously not ideal as a change is being flagged on every plan/apply.

Please advise.

Thanks.

**Comment by MaxxLiu22** [comment]
Hi @zippanto ,

Thank you for raising this issue, I can reproduce that, when internet_service = "enable", "service" and "dstaddr" blocks are not required, I have reported this to the development team. Sorry for any inconvenience.

Thanks,
Maxx

**Comment by MaxxLiu22** [comment]
Hi @zippanto 

This issue has been fixed in our latest Terraform FOS 1.19.0, these arguments are not required anymore. I will go ahead to close this case, you are free to reopen it or another case if you have any other questions.

Thanks,
Maxx
